src/retry.rs

```diff
@@ -39,7 +39,10 @@ where
     let func = Arc::new(Mutex::new(func));
     backoff::future::retry(
         ExponentialBackoff {
-            max_elapsed_time: Some(backoff_limit),
+            // We normally want to stop retrying after a while by setting max_elapsed_time.
+            // However, in order to work around unexplained transient failures, we want to
+            // keep retrying until successful (or the process is terminated).
```

Comment:
We need to be on [gitlab-runner-rs 0.7+](https://github.com/collabora/gitlab-runner-rs/commit/91074562862488615be0a1b01ac5a8af8a759f88) in order for job cancellation to actually work. (With that in place, I'd honestly rather just completely remove the backoff limits altogether.)

Reply:
That said the job timeout needs to be very long (the LLVM build did take more than 20h at some point, so in the past we had a 1d timeout) so it may be reasonable to have a shorter timeout for connection issues.

Reply:
don't confuse the gitlab level timeout with the running timing out on API requests to OBS.. 

The gitlab job timeout is driven by gitlab via the cancellation mechanism ; and so yes requires a bump to a newer gitlab-runner-rs (surprised that didn't happen yet) .  

This PR is purely for the runner timing outs waiting for the OBS API; which like @refi64 i would prefer to simply remove. There is no good timeout for that an admin can decide, so just using the "until gitlab cancels us" as the limit is the better option.